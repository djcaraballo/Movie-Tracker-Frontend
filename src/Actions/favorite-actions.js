export const addFavorite = (movie, user) => ({
  type: 'ADD_FAVORITE',
  movie_id: movie.id,
  user_id: user.id,
  title: movie.title,
  poster_path: movie.poster_path,
  release_date: movie.release_date,
  vote_average: movie.vote_average,
  overview: movie.overview,
});