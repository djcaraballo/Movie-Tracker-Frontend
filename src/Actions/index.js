export const addMovies = (movieInfo) => ({
  type: 'ADD_MOVIES',
  movies: movieInfo
})