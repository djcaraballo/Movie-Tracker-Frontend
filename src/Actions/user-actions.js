export const newUser = (name, email, password) => ({
  type: 'NEW_USER',
  name,
  email,
  password,
});

export const fetchUser = (status) => ({
  type: 'FETCH_USER',
  status,
});

export const toggleUserStatus = (status) => ({
  type: 'TOGGLE_STATUS',
  status: !status,
});
