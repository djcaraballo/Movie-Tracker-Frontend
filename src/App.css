.App {
  text-align: center;
  background: #000;
  /*color: #CFCFCD;*/
}

.title-container {
  /*background: #000;*/
  /*height: 10rem;*/
}

.app-title
{
  color: #F5C21B;
  background: -webkit-gradient(linear, left top, left bottom, from(#ECF0EF), to(#066168));
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: inline-block;
  font-family: "Montserrat", sans-serif;
  font-size: 7em;
  font-weight: 900;
  position: relative;
  text-transform: uppercase;
  text-shadow: 6px 0 #066168;
}

.app-title::before
{
  background-position: -180px;
  -webkit-animation: flare 5s infinite;
  -webkit-animation-timing-function: linear;
  background-image: linear-gradient(65deg, transparent 20%, rgba(255, 255, 255, 0.2) 20%, rgba(255, 255, 255, 0.3) 27%, transparent 27%, transparent 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  content: "MovieTracker";
  color: #FFF;
  display: block;
  padding-right: 140px;
  position: absolute;
  /*z-index: 2;*/

}

.app-title::after
{
  content: "MovieTracker";
  color: #FFF;
  display: block;
  position: absolute;
  text-shadow: 0 1px #563440, 0 2px #563440, 0 3px #563440, 0 4px #563440, 0 5px #563440, 0 6px #563440, 0 7px #563440, 0 8px #563440, 0 9px #563440, 0 10px #563440;
  top: 0;
  z-index: -1;
}

.sparkle::after, .sparkle::before
{
  -webkit-animation: sparkle 5s infinite;
  -webkit-animation-timing-function: linear;
  background: #FFF;
  border-radius: 100%;
  box-shadow: 0 0 5px #FFF, 0 0 10px #FFF, 0 0 15px #FFF, 0 0 20px #FFF, 0 0 25px #FFF, 0 0 30px #FFF, 0 0 35px #FFF;
  content: "";
  display: block;
  height: 10px;
  opacity: 0.7;
  position: absolute;
  width: 10px;
}

.sparkle::before
{
  -webkit-animation-delay: 0.2s;
  height: 7px;
  left: 0.10em;
  top: 0.9em;
  width: 7px;
}

.sparkle::after
{
  top: 0.32em;
  right: -5px;
}

@-webkit-keyframes flare
{
  0%   { background-position: -180px; }
  30%  { background-position: 700px; }
  100% { background-position: 700px; }
}

@-webkit-keyframes sparkle
{
  0%   { opacity: 0; }
  30%  { opacity: 0; }
  40%  { opacity: 0.8; }
  60%  { opacity: 0; }
  100% { opacity: 0; }
}

/*.title-container {
  height: 20em;
=======
  height: 12em;
>>>>>>> Add styling to login page
}

.app-title {
  color: #F5C21B;
  background: -webkit-gradient(linear, left top, left bottom, from(#ECF0EF), to(#563440));
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: inline-block;
  font-family: "Montserrat", sans-serif;
  font-size: 7em;
  font-weight: 900;
  position: relative;
  text-transform: uppercase;
}

.app-title::before
{
  background-position: -180px;
  -webkit-animation: flare 5s infinite;
  -webkit-animation-timing-function: linear;
  background-image: linear-gradient(65deg, transparent 20%, rgba(255, 255, 255, 0.2) 20%, rgba(255, 255, 255, 0.3) 27%, transparent 27%, transparent 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  content: "MovieTracker";
  color: #FFF;
  display: block;
  padding-right: 140px;
  position: absolute;
}

.app-title::after
{
  content: "MovieTracker";
  color: #FFF;
  display: block;
  position: absolute;
  text-shadow: 0 1px #00695C, 0 2px #00695C, 0 3px #00695C, 0 4px #00695C, 0 5px #00695C, 0 6px #00695C, 0 7px #00695C, 0 8px #00695C, 0 9px #00695C, 0 10px #00695C;
  top: 0;
  z-index: -1;
}

.inner-app-title::after, .inner-app-title::before
{
    -webkit-animation: sparkle 5s infinite;
  -webkit-animation-timing-function: linear;
  background: #FFF;
  border-radius: 100%;
  box-shadow: 0 0 5px #FFF, 0 0 10px #FFF, 0 0 15px #FFF, 0 0 20px #FFF, 0 0 25px #FFF, 0 0 30px #FFF, 0 0 35px #FFF;
  content: "";
  display: block;
  height: 10px;
  opacity: 0.7;
  position: absolute;
  width: 10px;
}

.inner-app-title::before
{
  -webkit-animation-delay: 0.2s;
  height: 7px;
  left: 0.05em;
  top: 0.9em;
  width: 7px;
}

.inner-app-title::after
{
  top: 0.3em;
  right: 0.075em;
}

@-webkit-keyframes flare
{
  0%   { background-position: -180px; }
  30%  { background-position: 800px; }
  100% { background-position: 800px; }
}

@-webkit-keyframes sparkle
{
  0%   { opacity: 0; }
  30%  { opacity: 0; }
  40%  { opacity: 0.8; }
  60%  { opacity: 0; }
  100% { opacity: 0; }
}*/


