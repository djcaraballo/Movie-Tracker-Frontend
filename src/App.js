import React, { Component } from 'react';
import * as DataCleaner from './Utils/Cleaners/';
import MovieContainer from './Containers/MovieContainer';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux'
import { addMovies } from './Actions'

class App extends Component {

  async componentDidMount() {
    const movieInfo = await DataCleaner.cleanMovieData();
    this.props.addMovies(movieInfo)
  }

  render() {
    return (
      <div className="App">
        <MovieContainer />
      </div>
    );
  }

}

const mapDispatchToProps = (dispatch) => {
  return {
    addMovies: (movieInfo) => {
      dispatch(addMovies(movieInfo))
    }
  }
}

export default connect(null, mapDispatchToProps)(App);
