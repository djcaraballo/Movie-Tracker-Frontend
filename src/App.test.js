import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { addMovies } from './Actions/';

describe('App', () => {
  let wrapper;
  let mockStore;
  const mockAddMovies = jest.fn()
  const mockUser = {
    name: "Louisa",
    id: 1806,
    email: "Louisa@turing.io",
    password: "oooooooooooookay"
  };

  beforeEach(() => {
    wrapper = shallow(<App 
      addMovies={mockAddMovies}
      user={mockUser} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToProps function", () => {
    it("should return an object with the user", () => {
      const mockState = {
        movies: [],
        user: { name: "Louisa", email: "blah@gmail.com" },
        favorites: []
      };
      const expected = {
        user: { name: "Louisa", email: "blah@gmail.com" }
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps function', () => {
    it('should call dispatch with an addMovies action when addMovies is called', () => {
      const mockMovieInfo = {
        backdrop_path:
          'http://image.tmdb.org/t/p/w1280/f4E0ocYeToEuXvczZv6QArrMDJ.jpg',
        id: 346910,
        overview:
          'From the outer reaches of space to the small-town streets of suburbia, the hunt comes home. Now, the universeâ€™s most lethal hunters are stronger, smarter and deadlier than ever before, having genetically upgraded themselves with DNA from other species. When a young boy accidentally triggers their return to Earth, only a ragtag crew of ex-soldiers and a disgruntled science teacher can prevent the end of the human race.',
        poster_path:
          'https://image.tmdb.org/t/p/w300_and_h450_bestv2/wMq9kQXTeQCHUZOG4fAe5cAxyUA.jpg',
        release_date: '2018-09-13',
        title: 'The Predator',
        vote_average: 5.3,
        favorite: false,
      }
      const mockDispatch = jest.fn()
      const mockAction = addMovies(mockMovieInfo)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addMovies(mockMovieInfo)
      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    })
  })
});
