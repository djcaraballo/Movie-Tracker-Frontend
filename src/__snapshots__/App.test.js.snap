// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(App)
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "subscribe": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addMovies": [Function],
      "store": Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "subscribe": [MockFunction],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addMovies": [Function],
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "subscribe": [MockFunction],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
